repo_branch.go

```diff
@@ -81,6 +81,12 @@ func (repo *Repository) GetBranches() ([]string, error) {
 	return branches, nil
 }
 
+// DeleteBranch delete a branch by name on repository.
```

Comment:
s/delete/deletes

Also, I think we should have `DeleteBranchOptions`:

```go
type DeleteBranchOptions struct {
    Force bool
}
```

In case of force deletion.

So the method signature becomes `DeleteBranch(name string, opts DeleteBranchOptions)`.